lib/chef/resource/hostname.rb

```diff
@@ -103,6 +128,10 @@ def updated_ec2_config_xml
         end
       end
 
+      def is_domain_joined?
+        powershell_exec!("(Get-WmiObject -Class Win32_ComputerSystem).PartofDomain").result
```

Comment:
should use `Get-CimInstance` over `Get-WmiObject`

Reply:
thanks. Adding that now